sql: fake a txn for logging COPY

otan opened the pull request:
Previously, txn for the event_log for COPY may be nil, meaning
the statement does not get logged in the event log. We previously relied
on assigning Timestamp from kv.Txn if the txn not nil, but this
triggers an assertion error (which is ignored) if that is the case.

Work around that by faking a txn during logging for copy instead.

Release justification: low-risk bug fix for logging

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/87094)
<!-- Reviewable:end -->

otan commented:
hmm wait this only affects older versions.

otan commented:
hmm, might as well for safety

otan commented:
this only gets hit if the first statement is COPY :|

otan committed:
sql: fake a txn for logging COPY

Previously, txn for the event_log for COPY may be nil, meaning
the statement does not get logged in the event log. We previously relied
on assigning Timestamp from kv.Txn if the txn not nil, but this
triggers an assertion error (which is ignored) if that is the case.

Work around that by faking a txn during logging for copy instead.

Release justification: low-risk bug fix for logging

Release note: None

ajwerner reviewed (commented):
This PR makes me apprehensive. We already have so many scary hacks around planner state. This is just going to add another. Why do we need the transaction?

otan commented:
the event log needs a transaction to log a timestamp -- if it is nil it does panic.
```
event.CommonDetails().Timestamp = txn.ReadTimestamp().WallTime
```

or otherwise it fails a later assertion: https://github.com/cockroachdb/cockroach/blob/1c96b93d9c5cdc213b48cc563b25d963ced5459c/pkg/sql/event_log.go#L518-L520

unfortunately COPY kind of abuses the planner in a sucky way and this feels "cleaner" to me then trying to plumb that is a COPY all the way down to `insertEventLog`.

otan commented:
actually i think i have a solution, let me change this!
